### PR TITLE
Broadcast tx can fail

### LINKED ISSUE #51: connect hangs if given an already-spent raw tx
lightningd gives the following error:

```
lightningd(12839): Broadcasting tx 020000000001015cdd29ac75ead0492ad22a4820f62abeef184f5fcefd50dff5aee9921ea6897d0000000017160014730569fc932bc0c9bd44e7a834d46bc938df4e61ffffffff0148e93c000000000022002009d9aa2eb572be5e33eea84f1e2241c88d9d9a8fb0665f79e78290eb880d271b0247304402205f7cc0f82bc6195ffefce0cfbd4daab62532074b6a09ca4513d91ba9603c3ffb02202afba691351b368f16b1a8e1d7f9597af33f9ab4f4768fb36cf2ef8988a12d3f01210318f97a9bce8e4a2c4cca0508219bf5e8839db02448572c729a22bb7c9198497000000000: error code: -26
error message:
18: bad-txns-inputs-spent
```

The connect json-rpc call should probably return right away and report this error to the user.

### PR DESCRIPTION
**Pull request by rustyrussell**
This should fix #51 but it's more of a refactor than I'd like.

### CODE REVIEW

File: daemon/bitcoind.c [hunk 1]
```diff
@@ -282,14 +282,13 @@ static void process_sendrawtx(struct bitcoin_cli *bcli)
 	cb(bcli->dstate, msg, bcli->cb_arg);
 }
 
-void bitcoind_sendrawtx_(struct peer *peer,
-			 struct lightningd_state *dstate,
+void bitcoind_sendrawtx_(struct peer *peer,struct lightningd_state *dstate,
```
Comment (cdecker) [inline daemon/bitcoind.c]: Minor nit: space between arguments

File: daemon/chaintopology.c [hunk 2]
```diff
@@ -237,7 +238,7 @@ static void try_broadcast(struct lightningd_state *dstate,
 	this_tx = txs[num_txs-1];
```
Comment (cdecker) [inline daemon/chaintopology.c]: This always confuses me, we are now sending `this_tx`, with the next call to `bitcoind_sendrawtx`, but it is also the transaction we report an expected error on?

File: daemon/chaintopology.c [hunk 3]
```diff
@@ -212,8 +213,8 @@ size_t get_tx_depth(struct lightningd_state *dstate,
 	return topo->tip->height - b->height + 1;
 }
 
-static void try_broadcast(struct lightningd_state *dstate,
-			  const char *msg, char **txs)
+static void broadcast_remainder(struct lightningd_state *dstate,
```
Comment (cdecker) [inline daemon/chaintopology.c]: This is indeed a better name for it :+1:

File: daemon/peer.c [hunk 4]
```diff
@@ -341,13 +340,16 @@ static void peer_breakdown(struct peer *peer)
 	
 	/* If we have a closing tx, use it. */
 	if (peer->closing.their_sig) {
+		const struct bitcoin_tx *close = mk_bitcoin_close(peer, peer);
 		log_unusual(peer->log, "Peer breakdown: sending close tx");
-		broadcast_tx(peer, bitcoin_close(peer), NULL);
+		broadcast_tx(peer, close, NULL);
+		tal_free(close);
```
Comment (cdecker) [inline daemon/peer.c]: Are we sure by this point that we don't need `close` anymore? Shouldn't `broadcast_tx` better take ownership of it?

Comment (cdecker) [inline daemon/peer.c]: Checking `broadcast_tx` it actually `tal_steal`s `close` and assigns it to a `struct outgoing_tx`, which after `tal_free` is a dangling pointer.

File: daemon/peer.c [hunk 5]
```diff
@@ -952,14 +954,17 @@ static bool closing_pkt_in(struct peer *peer, const Pkt *pkt)
...
 		/* BOLT #2:
 		 *
 		 * Once a node has sent or received a `close_signature` with
 		 * matching `close_fee` it SHOULD close the connection and
 		 * SHOULD sign and broadcast the final closing transaction.
 		 */
-		broadcast_tx(peer, bitcoin_close(peer), NULL);
+		close = mk_bitcoin_close(peer, peer);
+		broadcast_tx(peer, close, NULL);
+		tal_free(close);
```
Comment (cdecker) [inline daemon/peer.c]: Same as above.

File: daemon/chaintopology.c (hunk 2 above)
Comment (rustyrussell) [inline daemon/chaintopology.c]: You're right!  The error reporting is on the wrong tx.  Always was, it seems.  I'll append a patch which fixes this (leave the last tx on, then remove it before going to the next).

File: daemon/peer.c (hunk 4 above)
Comment (rustyrussell) [inline daemon/peer.c]: Actually, that was changed in 23107b7cdac5a57d265e8904bcdccdf4c6fbf4cc and this is just closing those leaks.  I will move the comment change to that commit, and document it better so this patch makes sense...

File: daemon/chaintopology.c (hunk 2 above)
Comment (rustyrussell) [inline daemon/chaintopology.c]: Actually, the first time it didn't remove the tx, so this was sending the first tx twice.  Erk, testing fix now.

**Comment by rustyrussell** [comment]
OK, those two fixed.

**Pull request closed without merging**